src/databricks/labs/remorph/reconcile/connectors/data_source_factory.py

```diff
@@ -0,0 +1,33 @@
+import logging
+
+from databricks.sdk import WorkspaceClient  # pylint: disable-next=wrong-import-order
+from pyspark.sql import SparkSession
+
+# pylint: disable-next=ungrouped-imports
+from databricks.labs.remorph.reconcile.connectors.data_source import DataSource
+from databricks.labs.remorph.reconcile.connectors.databricks import DatabricksDataSource
+from databricks.labs.remorph.reconcile.connectors.oracle import OracleDataSource
+from databricks.labs.remorph.reconcile.connectors.snowflake import SnowflakeDataSource
+from databricks.labs.remorph.reconcile.constants import SourceType
+
+logger = logging.getLogger(__name__)
+
+
+class DataSourceFactory:
+    @staticmethod
+    def get_data_source(
```

Comment:
this method is redundant here. move it as private method to `ReconConfigPrompts`

Reply:
this needs to be inside config.py/lib.py as a method which calls private dict which has this mapping and we do dict lookup check https://github.com/databrickslabs/remorph/blob/main/src/databricks/labs/remorph/config.py#L10

Reply:
moved to `config.py` but dict lookup won't be possible, because we need to initialize the class with Spark and WorkspaceClient

Reply:
Can't refactor `get_data_source` into `config.py` due to the `circular import` error, moved to `recon_config_utils.py`:

```
 File "remorph/src/databricks/labs/remorph/config.py", line 8, in <module>
    from databricks.labs.remorph.reconcile.connectors.data_source import DataSource
  File "remorph/src/databricks/labs/remorph/reconcile/connectors/data_source.py", line 7, in <module>
    from databricks.labs.remorph.config import TableRecon
ImportError: cannot import name 'TableRecon' from partially initialized module 'databricks.labs.remorph.config' (most likely due to a circular import) 
(remorph/src/databricks/labs/remorph/config.py)
Error: exit status 1
10:10:34 ERROR failed execution pid=68014 exit_code=1 error="exit status 1"
```